src/message-handler/src/NetworkAPIHandler.cpp

```diff
@@ -0,0 +1,16 @@
+#include "NetworkAPIHandler.h"
+
+NetworkAPIHandler::NetworkAPIHandler(IBSP& bsp, ILogger& logger) : m_bsp(bsp), m_logger(logger) {}
+
+std::optional<NetworkApiDTO> NetworkAPIHandler::handleApiCall(const NetworkApiDTO& apiCall) {
```

Comment:
I see, would it be possible instead, return an std::variant<ErrorEnum, std::optional<NetworkApiDTO>>.
So if you return an errorEnum, you know it's an error, if you return an empty optional, it's fine and no need to handle and if your optional has a network api, you then handle it. Alse the ErroEnum could be descriptive of what happend I guess. I think this would remove the need of a default constructor in back in propolis. What do you think?

Reply:
I like it. The distinction between the optional and monostate was not clear enough for which was an error and which was just that no call needs to be made afterwards. 

I am not sure how this will remove the need for the default constructor though since I will still return a NetworkApiDTO with a monostate to signify that no further call needs to be made